tpm2/kdf.go

```diff
@@ -0,0 +1,57 @@
+package tpm2
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"hash"
+)
+
+// KDFa implements TPM 2.0's default key derivation function, as defined in
+// section 11.4.9.2 of the TPM revision 2 specification part 1.
```

Comment:
I'm hesitant to do this; its a PDF so I can only link to the whole thing rather than the specific section, and I've already had the URLs break once since ive been working with them. Also, given the number of places we reference the spec, this could get heavy real fast. Are you sure?